core/unix/os_public.h

```diff
@@ -169,6 +169,9 @@ typedef kernel_sigcontext_t sigcontext_t;
 #    define SC_XSP SC_FIELD(sp)
 #    define SC_XFLAGS SC_FIELD(pstate)
 #    define SC_SYSNUM_REG SC_FIELD(regs[8])
+/* FIXME i#3544: RFC: Why is SC_RETURN_REG not defined for AArch64? It would
```

Comment:
Re RFC: I think x0 to x7 are used to return the result in the AArch64 calling convention: https://developer.arm.com/documentation/den0024/a/The-ABI-for-ARM-64-bit-Architecture/Register-use-in-the-AArch64-Procedure-Call-Standard/Parameters-in-general-purpose-registers. So we don't have a single register to define SC_RETURN_REG.

Reply:
Right, but those defines are for system calls and Linux's `syscall()` returns `long`, which is **usually** within word size. Note that `SC_RETURN_REG` is defined for `ARM` (line 195), even though the procedure call ABI specifies that [several registers may be used to return values](https://developer.arm.com/documentation/den0013/d/Application-Binary-Interfaces/Procedure-Call-Standard/Returning-results). In general the return value is written into the first of the return registers (`x0` on ARM, `a0` on RISC-V) when it fits (so anything up to word size) and the rest goes to the next register (i.e. 128bit return would go to `x0` and `x1`).

Reply:
Okay, I see on https://man7.org/linux/man-pages/man2/syscall.2.html that even x86 technically has two return regs, but os_public.h still defines SC_RETURN_REG.

Defining it for AArch64 sounds good to me.

Reply:
Done.